### ISSUE TITLE
Fix: remove redundant field in perfect hashjoin.

### ISSUE THREAD

**Issue by Light-City** [duckdb/duckdb#14085]
### What happens?

The is_probe_in_domain field in perfect hashjoin is redundant and invalid and needs to be deleted.

### To Reproduce

main branch.

### OS:

MacBook Pro (16-inch, 2019)

### DuckDB Version:

main

### DuckDB Client:

./duckdb 

### Full Name:

light

### Affiliation:

city

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a source build

### Did you include all relevant data sets for reproducing the issue?

No - Other reason (please specify in the issue body)

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Tishj** [comment]
Care to elaborate with your findings on why this is irrelevant?

**Comment by Tishj** [comment]
<https://github.com/duckdb/duckdb/pull/1959> this looks like an unimplemented TODO, reading the PR (bottom lines)

**Comment by Light-City** [comment]
> #1959

Got it. I'll close this issue.

**Comment by Tishj** [comment]
That's not what I mean, but providing some context around the variable, saying that it's never used, checking the commit history of what introduced it etc would have been helpful

The commit is from 3 years ago, it might be valid to just remove the leftover stubs from an optimization that was never finished. But it might also be an idea to finish the implementation of the optimization instead

**Comment by Light-City** [comment]
> That's not what I mean, but providing some context around the variable, saying that it's never used, checking the commit history of what introduced it etc would have been helpful
> 
> The commit is from 3 years ago, it might be valid to just remove the leftover stubs from an optimization that was never finished. But it might also be an idea to finish the implementation of the optimization instead
> 
> In any case it's definitely an oversight and a good find that this left over dead code exists 👍

Yes, I also saw this PR and found this unimplemented redundant code, so I raised this issue.

**Comment by Light-City** [comment]
> That's not what I mean, but providing some context around the variable, saying that it's never used, checking the commit history of what introduced it etc would have been helpful
> 
> The commit is from 3 years ago, it might be valid to just remove the leftover stubs from an optimization that was never finished. But it might also be an idea to finish the implementation of the optimization instead
> 
> In any case it's definitely an oversight and a good find that this left over dead code exists 👍

On the other hand I also submitted a code that uses this field, which could be optimized a bit.
